File: pset1/mario/mario.c
#include <cs50.h>
#include <stdio.h>

string pyramid(int height);

int main(void)
{
    int height;

    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    pyramid(height);

}

string pyramid(int height)
{

    const string hash = "#";
    const string space = " ";
    const int middle_spacing = 2;

    for (int i = 0; i < height; i++)
    {
        for (int j = 0; j < (height - (i + 1)); j++)
        {
            printf("%s", space); // Print L-H spaces
        }
        for (int j = 0; j < (i + 1); j++)
        {
            printf("%s", hash); // Print L-H hashes
        }
        for (int j = 0; j < middle_spacing; j++)
        {
            printf("%s", space); // Print middle spaces
        }
        for (int j = 0; j < (i + 1); j++)
        {
            printf("%s", hash); // Print R-H hashes
        }
        printf("\n");
    }
    return 0;
}

File: pset6/mario/more/mario.py
from cs50 import get_int


def main():

    # Almost an exact copy of the "c" version. I understand that I could
    # "abstractize" this more due to the high level nature of Python.
    # However I am busy this week and have less time to dedicate to this than I would like.

    while True:
        height = get_int("Positive Integer: ")
        if(height > 0 and height <= 8):
            break

    pyramid(height)


def pyramid(height):
    hash = "#"  # Hash can be changed by anything.
    space = " "  # Space can be replaced by anything
    middle_spacing = 2  # "adjustable" middle spacing

    # See the first comment.

    for i in range(height):
        for j in range(height - (i + 1)):
            print(f"{space}", end="")
        for j in range(i + 1):
            print(f"{hash}", end="")
        for j in range(middle_spacing):
            print(f"{space}", end="")
        for j in range(i + 1):
            print(f"{hash}", end="")
        print("")


main()